src/hivpy/experiment.py

```diff
@@ -71,7 +71,9 @@ def create_simulation(experiment_param):
 
 
 def create_log(log_param):
-    log_dir = log_param['log_directory']
+    log_dir = Path(log_param['log_directory'])
```

Comment:
Note that this means we pass a `Path` rather than a string to the `LoggingConfig` constructor, which contradicts the type hints for that argument. But it still works fine!

Reply:
It is fine in practice: `os.path.join` (which is what this `log_dir` is eventually passed to inside `LoggingConfig`) will accept any [path-like argument](https://docs.python.org/3/library/os.html#os.PathLike), whether it's a string or a `Path` object.

This is only introducing a minor inconsistency in the type hints of `LoggingConfig`, where we say that `log_dir` is a string. I think the easiest way around it is using an extra local variable for the `Path` inside `create_log` (and for creating the parents if needed), and passing in the string itself to `LoggingConfig`.

Alternatively, we could:
- change the type hint of `log_dir` in `LoggingConfig` to something more general, as suggested in [this answer](https://stackoverflow.com/questions/53418046/how-do-i-type-hint-a-filename-in-a-function)
- or switch over everything to use `Path`s instead of `os.path` functions, but that's more work!

Reply:
Easy way it is!